code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -365,12 +365,15 @@
 	if(!istype(W))
 		return 0
 
-	//Allow smashing of storage items on harm intent without also putting the weapon into the container.
-	var/mob/user = usr
-	if (user.a_intent == I_HURT && breakable_flags & BREAKABLE_WEAPON && loc != user)
-		return 0
+
 
 	if(usr) //WHYYYYY
+
+		//Allow smashing of storage items on harm intent without also putting the weapon into the container.
+		var/mob/user = usr
```

Comment:
Is there a point to this? `usr` is already a mob

Reply:
I was advised in another PR review to avoid directly referencing `usr` wherever possible. But maybe it would be better to avoid an unnecessary variable declaration?

Reply:
You need to avoid `usr` if you have the option of getting a reference to the mob you want as a parameter to your proc.
Rewriting `handle_item_insertion` and calls to it to have a `mob/user` parameter would be good, but out of scope.
Doing this cast would just add a line for no benefit.
The issue with `usr` is that it's a reference to the mob that started the current proc chain, so if *somehow* a player made another player insert something into a storage object, `usr` would not refer to the player who is actually performing the insertion. This has caused bugs in the past.